lib/services/db_userdata.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:relic_bazaar/model/user_model.dart';
+
+class DbUserData {
+  static final DbUserData _singleton = new DbUserData._internal();
+  DbUserData._internal();
+  static DbUserData get instance => _singleton;
+
+  UserModel user ;
+  
+  //used to fetch user information from firebase
+  Future<UserModel> fetchData() async {
```

Comment:
Receive the user id here, as we are already getting the authenticated user info from the streambuilder